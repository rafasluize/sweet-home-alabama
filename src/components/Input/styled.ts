import styled from 'styled-components';

export const Input = styled.div`
  margin-bottom: 24px;
  flex: 1;

  input {
    background: #ffffff;
    box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.25);
    border-radius: 2px;
    padding: 15px;
    display: block;
    width: 100%;
    border: none;
    height: 48px;

    &::placeholder {
      color: #909090;
    }
  }
`;

export default Input;
