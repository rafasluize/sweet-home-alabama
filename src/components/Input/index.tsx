import React, { FC, FormEvent } from 'react';
import * as Styled from './styled';

interface InputProps {
  name: string;
  label?: string;
  placeholder?: string;
  value?: string | number;
  type?: string;
  maxLength?: number;
  onChange(e: string): void;
}

const Input: FC<InputProps> = ({ label, name, value, onChange, placeholder, maxLength, type }) => (
  <Styled.Input>
    {label && <label htmlFor="name">{label}</label>}
    <input
      name={name}
      id={name}
      type={type}
      data-testid="test-input"
      onChange={(e: FormEvent<HTMLTextAreaElement | HTMLInputElement>) => onChange(e.currentTarget.value)}
      value={value}
      maxLength={maxLength}
      placeholder={placeholder}
    />
  </Styled.Input>
);

export default Input;
