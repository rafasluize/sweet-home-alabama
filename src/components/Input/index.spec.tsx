import React from 'react';

import { fireEvent, render } from '@testing-library/react';
import Input from '.';

const handle = jest.fn();

describe('<Input />', () => {
  test('snapshot', () => {
    const { container } = render(<Input name="teste" onChange={handle} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('onchange', () => {
    const { getByTestId } = render(<Input name="teste" onChange={() => handle} label="Teste" />);
    const input = getByTestId('test-input');

    fireEvent.change(input, { target: { value: 'Rafaela' } });
    expect((input as HTMLInputElement).value).toBe('Rafaela'); // need to make a change so React registers "" as a change
  });
});
