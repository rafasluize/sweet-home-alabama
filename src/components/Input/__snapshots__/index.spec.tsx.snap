// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> snapshot 1`] = `
.c0 {
  margin-bottom: 24px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 input {
  background: #ffffff;
  box-shadow: 0px 1px 4px rgba(0,0,0,0.25);
  border-radius: 2px;
  padding: 15px;
  display: block;
  width: 100%;
  border: none;
  height: 48px;
}

.c0 input::-webkit-input-placeholder {
  color: #909090;
}

.c0 input::-moz-placeholder {
  color: #909090;
}

.c0 input:-ms-input-placeholder {
  color: #909090;
}

.c0 input::placeholder {
  color: #909090;
}

<div
  class="c0"
>
  <input
    data-testid="test-input"
    id="teste"
    name="teste"
    value=""
  />
</div>
`;
